File: _includes/code/tutorials/tokenization/tokenize_endpoint.py
# TokenizeEndpointFreeform
import weaviate
from weaviate.classes.config import Tokenization, Configure

client = weaviate.connect_to_local()

# Ad-hoc tokenization with custom config
result = client.tokenization.text(
    text="The organic café crème blend",
    tokenization=Tokenization.WORD,
    analyzer_config=Configure.text_analyzer(
        ascii_fold=True,
        stopword_preset="en",
    ),
)

print(f"indexed: {result.indexed}")
print(f"query:   {result.query}")
# END TokenizeEndpointFreeform

# Test: accent folding converts café→cafe, crème→creme
assert "cafe" in result.indexed, f"Expected 'cafe' in indexed: {result.indexed}"
assert "creme" in result.indexed, f"Expected 'creme' in indexed: {result.indexed}"
# Test: stopword "the" is in indexed but not in query
assert "the" in result.indexed, f"Expected 'the' in indexed: {result.indexed}"
assert "the" not in result.query, f"Expected 'the' removed from query: {result.query}"
# Test: non-stopwords are in both
assert "organic" in result.indexed and "organic" in result.query

# TokenizeEndpointFreeformResult
"""
indexed: ['the', 'organic', 'cafe', 'creme', 'blend']
query:   ['organic', 'cafe', 'creme', 'blend']
"""
# END TokenizeEndpointFreeformResult

# TokenizeEndpointCustomPreset
# Define a named "fr" preset and reference it from analyzer_config.
# stopword_presets is mutually exclusive with stopwords — pass at most one.
result = client.tokenization.text(
    text="La Tasse Bleue et le Bol",
    tokenization=Tokenization.WORD,
    analyzer_config=Configure.text_analyzer(stopword_preset="fr"),
    stopword_presets={
        "fr": ["le", "la", "les", "un", "une", "des", "du", "de", "et"],
    },
)

print(f"indexed: {result.indexed}")
print(f"query:   {result.query}")
# END TokenizeEndpointCustomPreset

# Test: French stopwords are indexed but removed from query
assert "la" in result.indexed and "et" in result.indexed and "le" in result.indexed
assert "la" not in result.query and "et" not in result.query and "le" not in result.query
assert "tasse" in result.query and "bleue" in result.query and "bol" in result.query

# TokenizeEndpointCustomPresetResult
"""
indexed: ['la', 'tasse', 'bleue', 'et', 'le', 'bol']
query:   ['tasse', 'bleue', 'bol']
"""
# END TokenizeEndpointCustomPresetResult

# Setup: create collection for property-based tokenization example
from weaviate.classes.config import Property, DataType

client.collections.delete("TokenizeDemo")
client.collections.create(
    name="TokenizeDemo",
    inverted_index_config=Configure.inverted_index(
        stopword_presets={
            "fr": ["le", "la", "les", "un", "une", "des", "du", "de", "et"],
        },
    ),
    properties=[
        Property(
            name="name_fr",
            data_type=DataType.TEXT,
            tokenization=Tokenization.WORD,
            text_analyzer=Configure.text_analyzer(stopword_preset="fr"),
        ),
    ],
    vector_config=Configure.Vectors.self_provided(),
)

# TokenizeEndpointProperty
# Tokenize using an existing property's configuration
result = client.tokenization.for_property(
    collection="TokenizeDemo",
    property_name="name_fr",
    text="La Tasse Bleue et le Bol",
)

print(f"indexed: {result.indexed}")
print(f"query:   {result.query}")
# END TokenizeEndpointProperty

# Test: all words are indexed (stopwords are still stored)
assert "la" in result.indexed and "et" in result.indexed and "le" in result.indexed
assert "tasse" in result.indexed and "bleue" in result.indexed and "bol" in result.indexed
# Test: French stopwords are removed from query tokens
assert "la" not in result.query and "et" not in result.query and "le" not in result.query
# Test: non-stopwords remain in query tokens
assert "tasse" in result.query and "bleue" in result.query and "bol" in result.query
assert len(result.indexed) == 6  # all 6 words indexed
assert len(result.query) == 3    # only 3 non-stopwords in query

client.collections.delete("TokenizeDemo")
client.close()

# TokenizeEndpointPropertyResult
"""
indexed: ['la', 'tasse', 'bleue', 'et', 'le', 'bol']
query:   ['tasse', 'bleue', 'bol']
"""
# END TokenizeEndpointPropertyResult

File: _includes/code/tutorials/tokenization/tokenize_endpoint.ts
// Tutorial: Tokenization -> Tokenize endpoint - TypeScript examples
import assert from 'assert';

// TokenizeEndpointFreeform
import weaviate from 'weaviate-client';

const client = await weaviate.connectToLocal();

// Ad-hoc tokenization with custom config
let result = await client.tokenize.text('The organic café crème blend', 'word', {
  analyzerConfig: {
    asciiFold: true,
    stopwordPreset: 'en',
  },
});

console.log(`indexed: ${JSON.stringify(result.indexed)}`);
console.log(`query:   ${JSON.stringify(result.query)}`);
// END TokenizeEndpointFreeform

// Test: accent folding converts café->cafe, crème->creme
assert.ok(result.indexed.includes('cafe'), `Expected 'cafe' in indexed: ${result.indexed}`);
assert.ok(result.indexed.includes('creme'), `Expected 'creme' in indexed: ${result.indexed}`);
// Test: stopword "the" is in indexed but not in query
assert.ok(result.indexed.includes('the'), `Expected 'the' in indexed: ${result.indexed}`);
assert.ok(!result.query.includes('the'), `Expected 'the' removed from query: ${result.query}`);
// Test: non-stopwords are in both
assert.ok(result.indexed.includes('organic') && result.query.includes('organic'));

// TokenizeEndpointFreeformResult
/*
indexed: ['the', 'organic', 'cafe', 'creme', 'blend']
query:   ['organic', 'cafe', 'creme', 'blend']
*/
// END TokenizeEndpointFreeformResult

// TokenizeEndpointCustomPreset
// Define a named "fr" preset and reference it from analyzerConfig.
// stopwordPresets is mutually exclusive with stopwords - pass at most one.
result = await client.tokenize.text('La Tasse Bleue et le Bol', 'word', {
  analyzerConfig: { stopwordPreset: 'fr' },
  stopwordPresets: {
    fr: ['le', 'la', 'les', 'un', 'une', 'des', 'du', 'de', 'et'],
  },
});

console.log(`indexed: ${JSON.stringify(result.indexed)}`);
console.log(`query:   ${JSON.stringify(result.query)}`);
// END TokenizeEndpointCustomPreset

// Test: French stopwords are indexed but removed from query
assert.ok(result.indexed.includes('la') && result.indexed.includes('et') && result.indexed.includes('le'));
assert.ok(!result.query.includes('la') && !result.query.includes('et') && !result.query.includes('le'));
assert.ok(result.query.includes('tasse') && result.query.includes('bleue') && result.query.includes('bol'));

// TokenizeEndpointCustomPresetResult
/*
indexed: ['la', 'tasse', 'bleue', 'et', 'le', 'bol']
query:   ['tasse', 'bleue', 'bol']
*/
// END TokenizeEndpointCustomPresetResult

// Setup: create a collection that defines a custom French stopword preset
// and references it from the property's textAnalyzer.
await client.collections.delete('TokenizeDemo');
await client.collections.create({
  name: 'TokenizeDemo',
  invertedIndex: {
    stopwordPresets: {
      fr: ['le', 'la', 'les', 'un', 'une', 'des', 'du', 'de', 'et'],
    },
  },
  properties: [
    {
      name: 'name_fr',
      dataType: 'text' as const,
      tokenization: 'word' as const,
      textAnalyzer: { stopwordPreset: 'fr' },
    },
  ],
  vectorizers: weaviate.configure.vectors.selfProvided(),
});

// TokenizeEndpointProperty
// Tokenize using an existing property's configuration
const propResult = await client.tokenize.forProperty('TokenizeDemo', 'name_fr', 'La Tasse Bleue et le Bol');

console.log(`indexed: ${JSON.stringify(propResult.indexed)}`);
console.log(`query:   ${JSON.stringify(propResult.query)}`);
// END TokenizeEndpointProperty

// Test: all 6 words are indexed (stopwords are still stored)
assert.ok(propResult.indexed.includes('la') && propResult.indexed.includes('et') && propResult.indexed.includes('le'));
assert.ok(propResult.indexed.includes('tasse') && propResult.indexed.includes('bleue') && propResult.indexed.includes('bol'));
assert.equal(propResult.indexed.length, 6);
// Test: French stopwords are removed from query tokens
assert.ok(!propResult.query.includes('la') && !propResult.query.includes('et') && !propResult.query.includes('le'));
assert.ok(propResult.query.includes('tasse') && propResult.query.includes('bleue') && propResult.query.includes('bol'));
assert.equal(propResult.query.length, 3);

await client.collections.delete('TokenizeDemo');
await client.close();

// TokenizeEndpointPropertyResult
/*
indexed: ['la', 'tasse', 'bleue', 'et', 'le', 'bol']
query:   ['tasse', 'bleue', 'bol']
*/
// END TokenizeEndpointPropertyResult
